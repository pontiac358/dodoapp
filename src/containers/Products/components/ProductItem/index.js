import React from 'react';
import PropTypes from 'prop-types';
import styled from '@emotion/styled'
import Btn from '../../../../shared/Btn'
import CountControl from '../../../../widgets/CountControl'

const Container = styled.div`
    width: calc(25% - 45px);
    margin-right: 45px;
    margin-bottom: 64px;
    @media only screen and (max-width: 768px) {
        width: calc(50% - 45px);   
    }
    @media only screen and (max-width: 480px) {
        width: 100%;
        margin-right:0;   
    }
`

const ImageWrap = styled.div`
    display: flex;
    align-items: center;
    justify-content: center;
    margin-bottom:10px
`
const Img = styled.img`
    max-width: 250px;
    max-height: 250px;
    display:block;
    width:100%;
`
const Title = styled.div`
    color: #0c0c0c;
    font-size: 16px;
    margin-bottom: 30px;
    text-align: center;
`
Title.displayName = 'Title';

const Price = styled.div`
    color: #0c0c0c;
    font-size: 18px;
    margin-bottom: 30px;
    text-align: center;
`
Price.displayName = 'Price';

const BtnWrap = styled.div`
    display: flex;
    align-items: center;
    justify-content: center;
`



const ProductItem = ({ product, addToCart, hasInCart}) => {

    return (
        <Container>
            <Title>{ product.title }</Title>
             <ImageWrap>
                 <Img src={ product.img }/>
             </ImageWrap>
            <Price>{ product.price } р</Price>
            <BtnWrap>
                {
                    hasInCart ? <CountControl product={product}/> : <Btn onClick={addToCart} title={'В корзину'}/>
                }
            </BtnWrap>
        </Container>
    );
}

ProductItem.propTypes = {
    product: PropTypes.object,
    addToCart: PropTypes.func,
    hasInCart: PropTypes.object,
}

export default ProductItem;