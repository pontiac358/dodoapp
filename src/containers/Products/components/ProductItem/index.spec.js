import React from 'react'
import { shallow } from 'enzyme'
import ProductItem   from '../ProductItem'

describe('ProductItem component', () => {
    const props = {
        product: {
            title: 'test',
            img: '/test.png',
            price: 100
        },
        addToCart: () => {},
        hasInCart: {}
    };

    const wrapper = shallow(<ProductItem {...props} />);

    it("render correctly ProductItem container", () => {
        expect(wrapper).toMatchSnapshot();
    });

    it("render correctly product title prop", () => {
        expect(wrapper.find('Title').at(0).text()).toEqual(props.product.title);
    });

    it("render correctly product img prop", () => {
        expect(wrapper.find({ src: props.product.img })).toHaveLength(1)
    });

    it("render correctly product price prop", () => {
        expect(wrapper.find('Price').at(0).text()).toEqual(props.product.price + ' Ñ€');
    });
});

