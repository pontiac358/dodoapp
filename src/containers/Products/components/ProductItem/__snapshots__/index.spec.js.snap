// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductItem component render correctly ProductItem container 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ProductItem
    addToCart={[Function]}
    hasInCart={Object {}}
    product={
      Object {
        "img": "/test.png",
        "price": 100,
        "title": "test",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <ForwardRef(render)>
          test
        </ForwardRef(render)>,
        <ForwardRef(render)>
          <ForwardRef(render)
            src="/test.png"
          />
        </ForwardRef(render)>,
        <ForwardRef(render)>
          100
           р
        </ForwardRef(render)>,
        <ForwardRef(render)>
          <Connect(CountControl)
            product={
              Object {
                "img": "/test.png",
                "price": 100,
                "title": "test",
              }
            }
          />
        </ForwardRef(render)>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": "test",
        },
        "ref": null,
        "rendered": "test",
        "type": Object {
          "$$typeof": Symbol(react.forward_ref),
          "__emotion_base": "div",
          "__emotion_forwardProp": undefined,
          "__emotion_real": [Circular],
          "__emotion_styles": Array [
            "
    color: #0c0c0c;
    font-size: 16px;
    margin-bottom: 30px;
    text-align: center;
",
          ],
          "defaultProps": undefined,
          "displayName": "Title",
          "render": [Function],
          "withComponent": [Function],
        },
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": <ForwardRef(render)
            src="/test.png"
          />,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "src": "/test.png",
          },
          "ref": null,
          "rendered": null,
          "type": Object {
            "$$typeof": Symbol(react.forward_ref),
            "__emotion_base": "img",
            "__emotion_forwardProp": undefined,
            "__emotion_real": [Circular],
            "__emotion_styles": Array [
              "
    max-width: 250px;
    max-height: 250px;
    display:block;
    width:100%;
",
            ],
            "defaultProps": undefined,
            "displayName": "Styled(img)",
            "render": [Function],
            "withComponent": [Function],
          },
        },
        "type": Object {
          "$$typeof": Symbol(react.forward_ref),
          "__emotion_base": "div",
          "__emotion_forwardProp": undefined,
          "__emotion_real": [Circular],
          "__emotion_styles": Array [
            "
    display: flex;
    align-items: center;
    justify-content: center;
    margin-bottom:10px
",
          ],
          "defaultProps": undefined,
          "displayName": "Styled(div)",
          "render": [Function],
          "withComponent": [Function],
        },
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": Array [
            100,
            " р",
          ],
        },
        "ref": null,
        "rendered": Array [
          100,
          " р",
        ],
        "type": Object {
          "$$typeof": Symbol(react.forward_ref),
          "__emotion_base": "div",
          "__emotion_forwardProp": undefined,
          "__emotion_real": [Circular],
          "__emotion_styles": Array [
            "
    color: #0c0c0c;
    font-size: 18px;
    margin-bottom: 30px;
    text-align: center;
",
          ],
          "defaultProps": undefined,
          "displayName": "Price",
          "render": [Function],
          "withComponent": [Function],
        },
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": <Connect(CountControl)
            product={
              Object {
                "img": "/test.png",
                "price": 100,
                "title": "test",
              }
            }
          />,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "product": Object {
              "img": "/test.png",
              "price": 100,
              "title": "test",
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": Object {
          "$$typeof": Symbol(react.forward_ref),
          "__emotion_base": "div",
          "__emotion_forwardProp": undefined,
          "__emotion_real": [Circular],
          "__emotion_styles": Array [
            "
    display: flex;
    align-items: center;
    justify-content: center;
",
          ],
          "defaultProps": undefined,
          "displayName": "Styled(div)",
          "render": [Function],
          "withComponent": [Function],
        },
      },
    ],
    "type": Object {
      "$$typeof": Symbol(react.forward_ref),
      "__emotion_base": "div",
      "__emotion_forwardProp": undefined,
      "__emotion_real": [Circular],
      "__emotion_styles": Array [
        "
    width: calc(25% - 45px);
    margin-right: 45px;
    margin-bottom: 64px;
    @media only screen and (max-width: 768px) {
        width: calc(50% - 45px);   
    }
    @media only screen and (max-width: 480px) {
        width: 100%;
        margin-right:0;   
    }
",
      ],
      "defaultProps": undefined,
      "displayName": "Styled(div)",
      "render": [Function],
      "withComponent": [Function],
    },
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <ForwardRef(render)>
            test
          </ForwardRef(render)>,
          <ForwardRef(render)>
            <ForwardRef(render)
              src="/test.png"
            />
          </ForwardRef(render)>,
          <ForwardRef(render)>
            100
             р
          </ForwardRef(render)>,
          <ForwardRef(render)>
            <Connect(CountControl)
              product={
                Object {
                  "img": "/test.png",
                  "price": 100,
                  "title": "test",
                }
              }
            />
          </ForwardRef(render)>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": "test",
          },
          "ref": null,
          "rendered": "test",
          "type": Object {
            "$$typeof": Symbol(react.forward_ref),
            "__emotion_base": "div",
            "__emotion_forwardProp": undefined,
            "__emotion_real": [Circular],
            "__emotion_styles": Array [
              "
    color: #0c0c0c;
    font-size: 16px;
    margin-bottom: 30px;
    text-align: center;
",
            ],
            "defaultProps": undefined,
            "displayName": "Title",
            "render": [Function],
            "withComponent": [Function],
          },
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": <ForwardRef(render)
              src="/test.png"
            />,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "src": "/test.png",
            },
            "ref": null,
            "rendered": null,
            "type": Object {
              "$$typeof": Symbol(react.forward_ref),
              "__emotion_base": "img",
              "__emotion_forwardProp": undefined,
              "__emotion_real": [Circular],
              "__emotion_styles": Array [
                "
    max-width: 250px;
    max-height: 250px;
    display:block;
    width:100%;
",
              ],
              "defaultProps": undefined,
              "displayName": "Styled(img)",
              "render": [Function],
              "withComponent": [Function],
            },
          },
          "type": Object {
            "$$typeof": Symbol(react.forward_ref),
            "__emotion_base": "div",
            "__emotion_forwardProp": undefined,
            "__emotion_real": [Circular],
            "__emotion_styles": Array [
              "
    display: flex;
    align-items: center;
    justify-content: center;
    margin-bottom:10px
",
            ],
            "defaultProps": undefined,
            "displayName": "Styled(div)",
            "render": [Function],
            "withComponent": [Function],
          },
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": Array [
              100,
              " р",
            ],
          },
          "ref": null,
          "rendered": Array [
            100,
            " р",
          ],
          "type": Object {
            "$$typeof": Symbol(react.forward_ref),
            "__emotion_base": "div",
            "__emotion_forwardProp": undefined,
            "__emotion_real": [Circular],
            "__emotion_styles": Array [
              "
    color: #0c0c0c;
    font-size: 18px;
    margin-bottom: 30px;
    text-align: center;
",
            ],
            "defaultProps": undefined,
            "displayName": "Price",
            "render": [Function],
            "withComponent": [Function],
          },
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": <Connect(CountControl)
              product={
                Object {
                  "img": "/test.png",
                  "price": 100,
                  "title": "test",
                }
              }
            />,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "product": Object {
                "img": "/test.png",
                "price": 100,
                "title": "test",
              },
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": Object {
            "$$typeof": Symbol(react.forward_ref),
            "__emotion_base": "div",
            "__emotion_forwardProp": undefined,
            "__emotion_real": [Circular],
            "__emotion_styles": Array [
              "
    display: flex;
    align-items: center;
    justify-content: center;
",
            ],
            "defaultProps": undefined,
            "displayName": "Styled(div)",
            "render": [Function],
            "withComponent": [Function],
          },
        },
      ],
      "type": Object {
        "$$typeof": Symbol(react.forward_ref),
        "__emotion_base": "div",
        "__emotion_forwardProp": undefined,
        "__emotion_real": [Circular],
        "__emotion_styles": Array [
          "
    width: calc(25% - 45px);
    margin-right: 45px;
    margin-bottom: 64px;
    @media only screen and (max-width: 768px) {
        width: calc(50% - 45px);   
    }
    @media only screen and (max-width: 480px) {
        width: 100%;
        margin-right:0;   
    }
",
        ],
        "defaultProps": undefined,
        "displayName": "Styled(div)",
        "render": [Function],
        "withComponent": [Function],
      },
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
